hnn_core/network.py

```diff
@@ -219,47 +293,443 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
-    def _add_external_feeds(self):
-        """Legacy function, for backward compatibility with original HNN-GUI
+    def add_evoked_drive(self, name, *,
+                         mu=None, sigma=None, numspikes=None,
+                         weights_ampa=None, weights_nmda=None, location=None,
+                         space_constant=3., dispersion_time=0.1, seedcore=42):
+        """Add an 'evoked' external drive to the network
 
-        This function is called exactly once during initialisation
+        Parameters
+        ----------
+        name : str
+            Unique name for drive
+        mu : float
+            Mean of Gaussian event time distribution (across target cells)
+        sigma : float
+            Standard deviation of event time distribution (across target cells)
+            Set to zero for "synchronous" evoked drive.
+        numspikes : int
+            Number of spikes at each target cell
+        weights_ampa : dict or None
+            Synaptic weights of AMPA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        weights_nmda : dict or None
+            Synaptic weights of NMDA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        space_constant : float
+            Describes lateral dispersion (from column origin) of synaptic
+            delays within the simulated column
+        dispersion_time : float or dict
+            Synaptic delay at the column origin, dispersed laterally as a
+            function of the space_constant. If float, applies to all target
```

Comment:
how about
```suggestion
            Maximum synaptic delay. If float, applies to all target
```
so users don't have to worry about the gory details of positions of the cell.

Reply:
But was does "maximum" (should be "minimum" btw) here refer to? Relative to what?

I agree about gory, but it is all detailed in the reference paper...

Reply:
looking at at [this](https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/cell.py#L353) shouldn't it have the maximum when d=0 which happens at the origin?

Reply:
If `nc['A_delay'] = 1`, then

<img src="https://render.githubusercontent.com/render/math?math=%241%20%2F%20e%5E%7B-(d%3D0)%5E2%2F%5Clambda%5E2%7D%20%3D%201%20%3C%201%20%2F%20e%5E%7B-(d%3E0)%5E2%2F%5Clambda%5E2%7D%24">

Note the minus-sign inside the exponent: the larger the d, the smaller the divisor.

Reply:
ufff ... I missed the division because of how the line is broken. Thanks for clarifying!